.github/workflows/deploy_stage.yaml

```
@@ -31,25 +33,99 @@ jobs:
     steps:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
+
       - name: Checkout
         uses: actions/checkout@v3
         with:
           ref: develop
+
       - name: Login to GitHub Container Registry
         uses: docker/login-action@v2
         with:
           registry: ${{ env.REGISTRY }}
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
+
       - name: Sanitize repo slug
         uses: actions/github-script@v6
         id: repo_slug
         with:
           result-encoding: string
           script: return 'ghcr.io/${{ github.repository }}'.toLowerCase()
+
       - name: Push Stage to GitHub Container Registry
         uses: docker/build-push-action@v4
         with:
           context: "{{defaultContext}}"
           push: true
           tags: ${{ steps.repo_slug.outputs.result }}:stage
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: build-and-push-image-to-github-packages
+    environment:
+      name: stage_deploy
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Copy docker-compose files
+        uses: appleboy/scp-action@master
+        with:
+          host: ${{ secrets.HOST }}
+          username: ${{ secrets.USERNAME }}
+          key: ${{ secrets.SSH_PRIVATE_KEY }}
+          source: "infra/docker-compose.prod.yml"
+          target: ${{ env.DEPLOY_PATH }}
+
+      - name: Create .env file
+        uses: appleboy/ssh-action@master
+        with:
+          host: ${{ secrets.HOST }}
+          ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
```

[Смотри документацию](https://github.com/appleboy/ssh-action?tab=readme-ov-file#input-variables)
```suggestion
          key: ${{ secrets.SSH_PRIVATE_KEY }}
```